### ISSUE TITLE
carousel transition speed

### ISSUE THREAD

**Issue by mesutuca** [fancyapps/ui#492]
### Describe the bug

When the slider forward button is clicked repeatedly, it flows extremely fast, I want to slow it down. such as the example; If the user pressed the forward button even 5 times, it should not wait, that is, almost all of them.
`
new Carousel(document.getElementById('pageCarousel'),
      {
        friction: 0.86,
        slidesToSlide: 1,
        Dots: true,
        center: true,
        fill: false,
        infinite: true,
        slidesPerPage: 1,
        transition: "slide",
        Autoplay: {
          timeout: 3000,
          pauseOnHover: true
        },
        preload: 1,
      },
    );
`

### Reproduction

Prevent clicking next without loading the 2nd image when next is done

**Comment by fancyapps** [comment]
Hi,

Sorry, but what do you man by `Prevent clicking next without loading the 2nd image when next is done`? I do not really understand this question.

**Comment by mesutuca** [comment]
If you watch the photo I have uploaded, you will understand.
I don't want the transitions to be that fast. I want to prevent the transition before the next photo is shown when the next slide is clicked.

[google foto](https://photos.app.goo.gl/zjSZr3P6z2y7xcKu6)

**Comment by mesutuca** [comment]
repetitive click next slide or prev slide is progressing extremely fast when clicked. Is there any way to prevent this?

**Comment by fancyapps** [comment]
It has nothing to do with speed. For example, if you double-click when the first slide is open, the second slide will be activated after the first click, and the third slide after the second. Accordingly, the carousel will be animated to the appropriate position. So it looks like you want to prevent navigation to the third slide while the carousel is still animating to the second slide. Sorry, there is currently no option for that

**Comment by fancyapps** [comment]
Hi,

You can use `beforeChange` event (available from v5.0.24) to prevent navigation if the transition has not yet completed:

```js
Fancybox.bind('[data-fancybox="gallery"]', {
  Carousel: {
    transition: "slide",
    on: {
      beforeChange: (carousel, event) => {
        if (carousel.inTransition.size) {
          event.preventDefault();
        }
      },
    },
  },
});    
```

https://jsfiddle.net/ohzgjf40/
